### DISCUSSION TITLE
Can Shopify product objects be extended?

### DISCUSSION THREAD

**Discussion by dungle-scrubs** [statamic-rad-pack/shopify#23] [Help] [answered]
This is actually a big one for me and I'm currently looking for a solution.

In normal Shopify builds we usually rely on metafield apps to append extra data to Shopify objects (product, variant, collection, etc).  

Is it a current feature to append additional types of information to a product object within Statamic, eliminating the need for Shopify metafields?

Info types such as: text, textarea, image, etc.

**Comment by jackabox** [comment, marked as answer]
Yes, they can be extended. They use collections under the hood. This lets you append any additional fields to either the product or variants. Therefore, you can append any field in Statamic to it by editing the blueprint. 

You wouldn't need to use meta fields and it's one of the key reasons I built this to be extendable and easier to control :)

**Reply by dungle-scrubs** [reply]
Amazing :) Do you think this would be manageable for a ship with about 10k products?

**Reply by jackabox** [reply]
Disclosure: I've tested this on 1000 products, but never 10k.

Initial import would be heavy, but make sure it runs on the queue.

In theory, it will work yes. I want to add a webhook for checking when a product is updated and pushing that change to the app - this should mean you only have to run the initial sync of all products once, then it can be done 1 by 1.

**Reply by jackabox** [reply]
FYI: added two new webhooks for v.1.2.0 that can be set up for product created and updated :)

**Reply by dungle-scrubs** [reply]
Awesome, this might be helpful for you as well. Shopify's [BulkOperation](https://shopify.dev/docs/admin-api/graphql/reference/bulk-operations/bulkoperation#fields-2021-01) might be helpful for a future release when it comes to large inventory shops. 

The Gatsby team released [video](https://www.youtube.com/watch?v=43fJTO9gMUE) demonstrating how it helped them with 10k product shops for the new version of their source plugin.

That, combined with extending objects with the metafields at the CMS level could be pretty solid.

**Reply by jackabox** [reply]
We don't need bulk operation. We can't insert them quicker into the data. The products are all queued on initial load and don't need to be again, as you can call products 1by1

**Reply by dungle-scrubs** [reply]
Thx, that makes sense now.
